dags/ethereumetl_airflow/build_parse_logs_dag.py

```diff
@@ -0,0 +1,168 @@
+
+dags_folder = os.environ.get('DAGS_FOLDER', '/home/airflow/gcs/dags')
+
+def abi_to_event_topic(abi):
+    return '0x' + event_abi_to_log_topic(abi).hex()
+
+def build_parse_logs_dag(
+    dag_id,
+    destination_dataset_project_id,
+    copy_dataset_project_id=None,
+    copy_dataset_name=None,
+    chain='ethereum',
+    notification_emails=None,
+    load_start_date=datetime(2018, 7, 1),
+    schedule_interval='0 0 * * *',
+    load_all_partitions=True
+):
+
+    source_dataset_name = f'crypto_{chain}'
+
+    if not destination_dataset_project_id:
+        raise ValueError('destination_dataset_project_id is required')
+
+    environment = {
+        'source_dataset_name': source_dataset_name,
+        'destination_dataset_project_id': destination_dataset_project_id,
+        'load_all_partitions': load_all_partitions
+    }
+
+    default_dag_args = {
+        'depends_on_past': False,
+        'start_date': load_start_date,
+        'email_on_failure': True,
+        'email_on_retry': False,
+        'retries': 5,
+        'retry_delay': timedelta(minutes=5)
+    }
+
+    dag = models.DAG(
+        dag_id,
+        catchup=False,
+        schedule_interval=schedule_interval,
+        default_args=default_dag_args)
+
+    def submit_bigquery_job(job, configuration):
+        try:
+            logging.info('Creating a job: ' + json.dumps(configuration.to_api_repr()))
+            result = job.result()
+            logging.info(result)
+            assert job.errors is None or len(job.errors) == 0
+            return result
+        except Exception:
+            logging.info(job.errors)
+            raise
+
+    def get_parse_logs_sql_template():
+        filepath = os.path.join(dags_folder, 'resources/stages/parse/sqls/parse_logs.sql')
+        with open(filepath) as file_handle:
+            content = file_handle.read()
+            return content
+
+    def get_list_of_json_files():
+        folder = os.path.join(dags_folder, 'resources/stages/parse/table_definitions/')
+        return [f for f in glob(folder + '*/*.json')]
+
+    def read_json_file(filepath):
+        with open(filepath) as file_handle:
+            content = file_handle.read()
+            return json.loads(content)
+
+    def create_struct_string_from_schema(schema):
+        return ', '.join([f.get('name') + ' ' + f.get('type') for f in schema])
+
+    def read_bigquery_schema_from_dict(schema):
+        result = []
+        for field in schema:
+            result.append(bigquery.SchemaField(
+                name=field.get('name'),
+                field_type=field.get('type', 'STRING'),
+                mode=field.get('mode', 'NULLABLE'),
+                description=field.get('description')))
+        return result
+
+    def create_task_and_add_to_dag(task_config):
+        dataset_name = task_config['table']['dataset_name']
+        table_name = task_config['table']['table_name']
+        table_description = task_config['table']['table_description']
```

Comment:
table_description not set to job_config.

Reply:
Done. I couldn't find a way to set table description directly in the `job_config`, but I do a separate `update_table` after the query has completed.